packages/babel-core/src/config/config-chain.js

```diff
@@ -316,7 +316,7 @@ function babelrcLoadEnabled(
 
   return babelrcPatterns.some(pat => {
     if (typeof pat === "string") {
-      pat = pathPatternToRegex(pat, babelrcRootsDirectory);
+      pat = pathPatternToRegex(pat, babelrcRootsDirectory, false);
```

Comment:
To be consistent we should also update

https://github.com/babel/babel/blob/08c7280167a8dd7696c16ac70e36d5d3120962a9/packages/babel-core/src/config/files/configuration.js#L297

Reply:
We can/should remove the default param value then.

Reply:
There is also a bunch of other places that eventually end up using `pathPatternToRegex`, for example `configIsApplicable`:

https://github.com/babel/babel/blob/08c7280167a8dd7696c16ac70e36d5d3120962a9/packages/babel-core/src/config/config-chain.js#L745-L758

On one hand, it makes ignore patterns more expressive because it makes `foo/*` act differently to `foo/**`, but it also breaks the `.gitignore`-style matching where directories are ignored recursively, for example `node_modules` ignores everything inside the `node_modules` directory.

This change seems like it has the potential to impact a lot more users.

Reply:
I think we shouldn't change it: except for `babelrcRoots`, everything else _should_ match nested directories imo.

Reply:
That is the current behaviour in this PR.